### ISSUE TITLE
Long loading time on multi-project

### ISSUE THREAD

**Issue by huajiang-tubi** [scalacenter/sbt-version-policy#205]
We got ~140 more seconds on sbt reload when sbt-version-policy is added to our multi-project with 10 libraries under the root. The problem turned out to be that, the [versionPolicyPreviousVersions](https://github.com/scalacenter/sbt-version-policy/blob/main/sbt-version-policy/src/main/scala/sbtversionpolicy/SbtVersionPolicyMima.scala#L92C5-L92C34) setting is resolved at load time by searching the resolvers for previous versions of the project artifacts. It take ~14 seconds for each subproject. Typically we work on a single project at a time, that makes fetching previous versions for all the other projects a waste. 

A potential optimization I'm thinking about is to make `versionPolicyPreviousVersions` a taskKey instead of settingKey, so the versions will be fetched only when actually needed. It would also need to make settingKeys depending on it taskKeys. I'm not sure how much work that would be, and is there any other thing could be done to improve?

**Comment by julienrf** [comment]
Hello, thank you for explaining the issue. I agree that 140 seconds is way too long. A workaround could be to manually set `versionPolicyPreviousVersions` so that it is not computed at build-load-time.

> It would also need to make settingKeys depending on it taskKeys.

Would that even be possible?

> It take ~14 seconds for each subproject. […] Another issue I found is that it fails silently without trying sbt credentials for private repositories, so we waited a long time and got nothing.

It seems this is the heart of the issue. Unfortunately, [because it is not possible to make a setting depend on task](https://github.com/scalacenter/sbt-version-policy/blob/ce2500efc9cef4b3f26a26b314f7951f5d4a192b/sbt-version-policy/src/main/scala/sbtversionpolicy/SbtVersionPolicyMima.scala#L44-L45), we do not “inherit” from the standard sbt configuration to resolve dependencies, which would explain why your credentials are not taken into account. The only solution I see to this problem would be to introduce new settings to configure the dependency resolution engine [here](https://github.com/scalacenter/sbt-version-policy/blob/ce2500efc9cef4b3f26a26b314f7951f5d4a192b/sbt-version-policy/src/main/scala/sbtversionpolicy/SbtVersionPolicyMima.scala#L30-L52). From what I have seen in your [draft PR](https://github.com/scalacenter/sbt-version-policy/pull/206/files), you only need to introduce a key to define the list of repositories to search in, which should have a default value corresponding to the current behavior. How does that sound? Please let me know if you would like more guidance.
